trin-core/src/utp/stream.rs

```diff
@@ -111,30 +141,59 @@ impl UtpListener {
                     PacketType::Syn => {
                         if let Some(enr) = self.discovery.discv5.find_enr(node_id) {
                             // If neither of those cases happened handle this is a new request
-                            let mut conn = UtpSocket::new(Arc::clone(&self.discovery), enr.clone());
-                            let _ = conn.handle_packet(&packet, enr);
+                            let mut conn = UtpSocket::new(
+                                Arc::clone(&self.discovery),
+                                enr.clone(),
+                                self.rx.clone(),
```

Comment:
I was under the impression that `clone`s in Rust is not ideal, and should be used only when dealing with static data. Doesn't seem like a receive channel is a great candidate for cloning

Reply:
This should be `Arc::clone(&self.rx)`.

> Doesn't seem like a receive channel is a great candidate for cloning

I'm not sure what do you exactly mean, could you elaborate? I think it is not unusual to clone a receiver channel as long as it is inside a lock, for example: `Arc<RwLock<T>>`.